test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/CompleteCommand.cs

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.DotNet.Cli.Utils;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.DotNet.Tools.Test.Utilities
+{
+    public sealed class CompleteCommand : DotnetCommand
```

Comment:
As always. I am not a big fan of shell level test. CompleteCommand has only about 100 lines. But these tests dup parser tests heavily. If I changed a parser, I need to change parser test and auto complete test. But they are different concern.

I think you should be able to just create a CompleteCommand object in memory and only test the logic in these 100 lines. Or just keep one of these shell level test just to make sure dotnet complete is plugged in correctly.

Some of the bugs are parser config out of sync, to repo it, they should be parser tests.

Reply:
I would need to refactor `CompleteCommand` to accept a logger since right now it uses `Console.WriteLine` and we can't capture the output if we don't spawn it as an external process.

I'm not opposed to making that change.

Reply:
sounds good. `public static int Run(string[] args) ` is the contact. So i this case you cannot add it to the interface. Extract a internal method should be easier. This is test induced damage. But the root cause is the contract of buildincommand ---  `public static int Run(string[] args) `

Reply:
(more on Run(), ideally build in commands should have a common interface. And program.cs should discover them by that. But from my understanding, in DNX days all build commands are actual different process with individual project.json. So the only contact we have is `public static int Run(string[] args)` and it stays)